### PR TITLE
334 combine highlyservd bus train

### PR DESCRIPTION
**Pull request by james-westwood**
## Pull Request submission 

*Insert detailed bullet points about your changes here!*

*Insert any instructions to help the reviewer, e.g. "install new requirements from `requirements.txt`"*

*Let the reviewer know what data files are needed (to be grabbed from sync)

#### Closes or fixes

Fixes #

#### Code

- [ ] **Requirements** My/our code matches the requirements of the ticket
- [ ] **Functionality**: New functions meet requirements in issue ticket
- [ ] **Compliant Code** Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
- [ ] **Code runs** The code runs on my machine
- [ ] **Clean Code** 
    - [ ] Code has been linted (use your favourite linter)
    - [ ] Code adheres to [DRY]([url](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)) 

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s') definition(s) have been created
    - [ ] Includes `parameters` and `returns` for all major functions 
    - [ ] Includes data types
- [ ] **Updated Documentation**: Working doc has been updated and a new ticket to update the master documentation has been created. 

#### Data
- [ ] All data needed to run this script has been added to Sync

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `requirements.txt`
- [ ] Documentation has been created and is clear - check the google Doc
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

**Comment by james-westwood** [comment]
@nkshaw23 I have created a draft PR to collect all the commits together and work from here. Might be best to keep comments about the work in the issue #334 though.

**Comment by nkshaw23** [comment]
Tested for England, results viewed in QGIS. Everything looks good but would like to investigate a bit further.

@james-westwood outstanding task for this PR:

1) Build in the process for Scotland script. In an ideal world we would have these combined into one process, but we can look at this afterwards.
2) Once in scotland, can remove two functions from data_transform (`add_stop_capacity_type` and `filter_stops`)
3) Check for other functions that are no longer used

**Comment by jwestw** [comment]
@nkshaw23 I think it would be best to split out the Scotland stuff into it's own ticket + PR.

**Comment by nkshaw23** [comment]
@james-westwood This is now ready for review as we are splitting Scotland work into another ticket.

File: config.yaml [hunk 1]
```diff
@@ -47,6 +47,8 @@ day_filter: 'general' #exact
 train_msn_filename: 'ttisf467.msn'
 train_mca_filename: 'ttisf467.mca'
 station_locations: 'station_locations.csv'
+bus_in_dir : 'data/england_bus_timetable/'
+train_in_dir : 'data/england_train_timetable'
```
Comment (nkshaw23) [inline config.yaml:51]: Does it matter that `train_in_dir` doesn't finish with a `/`?

**Comment by nkshaw23** [comment]
Pipeline works from start to finish now. Just two comments from me.

Because we cant join NAPTAN onto the train data, we couldnt use the existing `capacity_type` function which relied on `StopTypoe` codes from NAPTAN. Hence, the `high/low` capacity type has been added manually based on whether stop is bus, train or tram

**Comment by jwestw** [comment]
I will pull the code and run it today.

File: SDG_bus_timetable.py [hunk 2]
```diff
@@ -228,15 +235,16 @@
...
-highly_serviced_bus_stops_df = highly_serviced_bus_stops_df[list(
-    config["NAPTAN_TYPES"].keys())]
+# Drop the hours columns
+highly_serviced_bus_stops_df = (
+    highly_serviced_bus_stops_df['NaptanCode', 'Easting', 'Northing']
+   )
 
 # Save a copy to be ingested by SDG_11.2.1_main
 highly_serviced_bus_stops_df.to_feather(os.path.join(
-    output_directory, 'highly_serviced_stops.feather'))
```
[the remark this answers is not in the document]
Comment (jwestw) [reply SDG_bus_timetable.py:237]: Yes.

File: config.yaml (hunk 1 above)
Comment (jwestw) [reply config.yaml:51]: Not sure, I think not, but I will make it consistent.

File: data_transform.py [hunk 3]
```diff
@@ -195,35 +195,6 @@ def filter_stops(stops_df):
     return filter_stops
 
 
-def add_stop_capacity_type(stops_df):
```
Comment (jwestw) [inline data_transform.py:198]: Removing this is definitely wrong as it's an essential part of the method. Needs to be reverted

File: SDG_11.2.1_main.py [hunk 4]
```diff
@@ -33,31 +33,90 @@
...
+# Metros and trains added from NAPTAN as we dont have timetable
+# data for these stops. Hence, they wont be highly serviced.
+
+highly_serviced_bus_stops = di._feath_to_df('highly_serviced_stops',
+                                            BUS_IN_DIR)
+highly_serviced_train_stops = di._feath_to_df('train_highly_serviced_stops',
+                                              TRAIN_IN_DIR)
 
-# adds in high/low capacity column
-stops_geo_df = dt.add_stop_capacity_type(stops_df=stops_geo_df)
```
Comment (jwestw) [inline SDG_11.2.1_main.py:58]: Removing this is definitely wrong as it's an essential part of the method. Needs to be reverted

**Review by jwestw — CHANGES_REQUESTED** [review]
I wanted to run these scripts first (as they were altered)
- SDG_bus_timetable.py
- SDG_train_timetable.py

However, the bus script ends with "Killed". See image:

image

I want to debug this, but when I run it in the debugger, it runs fine and doesn't end with "Killed". Weird. Do you see something similar?

File: data_transform.py (hunk 3 above)
Comment (nkshaw23) [reply data_transform.py:198]: Popped back in as placeholder

**Review by jwestw — APPROVED** [review]

**Pull request merged**